flex-checks/src/main/java/org/sonar/flex/checks/XPathCheck.java

```diff
@@ -76,7 +76,7 @@ public void visitFile(@Nullable AstNode fileNode) {
         if (object instanceof AstNode) {
           AstNode astNode = (AstNode) object;
           addIssueAtLine(message, astNode.getTokenLine());
-        } else if (object instanceof Boolean && (Boolean) object) {
+        } else if (object instanceof Boolean && Boolean.TRUE.equals(object)) {
```

Comment:
`instanceof` is not needed, equals will cover it

Reply:
Changed